inventory/internal/store/instance.go

```diff
@@ -592,3 +614,11 @@ func isNotValidLocalAccountUpdate(
 	return slices.Contains(fieldmask.GetPaths(), instanceresource.EdgeLocalaccount) &&
 		instanceq.CurrentState != instanceresource.CurrentStateINSTANCE_STATE_UNSPECIFIED
 }
+
+func isNotValidCustomConfigUpdate(
+	fieldmask *fieldmaskpb.FieldMask,
+	instanceq *ent.InstanceResource,
+) bool {
+	return slices.Contains(fieldmask.GetPaths(), instanceresource.EdgeCustomConfig) &&
+		instanceq.CurrentState != instanceresource.CurrentStateINSTANCE_STATE_UNSPECIFIED
```

Comment:
As discussed, extract the condition for checking if a Instance is Provisioned/inProvisioning so it can be re-used for both this and localAccounts (and for others). Also, we should check the statusIndicator too.